Tests fail if ipv6 disabled

### Expected behavior

Tests should pass regardless of IP stack enabled.

### Actual behavior

The TestMain_Set_Quick test fails for trying to make a connection to http://[::]/...

### Steps to reproduce the behavior

Starting with Ubuntu 16.04.1 LTS with Go 1.8.

Append the following to /etc/sysctl.conf:

    net.ipv6.conf.all.disable_ipv6 = 1
    net.ipv6.conf.default.disable_ipv6 = 1
    net.ipv6.conf.lo.disable_ipv6 = 1

Run `sudo sysctl -p`

Run `go test github.com/pilosa/pilosa/cmd/pilosa`

See that the tests fail now:

    ubuntu@ip-10-0-1-239:~/gopath/src/github.com/pilosa/pilosa$ go test github.com/pilosa/pilosa/cmd/pilosa
    --- FAIL: TestMain_Set_Quick (0.00s)
    panic: Post http://[::]:44358/query?db=d: dial tcp [::]:44358: connect: cannot assign requested address [recovered]
            panic: Post http://[::]:44358/query?db=d: dial tcp [::]:44358: connect: cannot assign requested address

    goroutine 7 [running]:
    testing.tRunner.func1(0xc42005b1e0)
            /home/ubuntu/go/src/testing/testing.go:622 +0x29d
    panic(0x8a80e0, 0xc42011e510)
            /home/ubuntu/go/src/runtime/panic.go:489 +0x2cf
    github.com/pilosa/pilosa/cmd/pilosa_test.MustDo(0x91d455, 0x4, 0xc4200d7220, 0x1c, 0xc42011e270, 0x28, 0x0)
            /home/ubuntu/gopath/src/github.com/pilosa/pilosa/cmd/pilosa/main_test.go:403 +0x139
    github.com/pilosa/pilosa/cmd/pilosa_test.(*Main).Query(0xc42007b180, 0x91d625, 0x4, 0xc42011e270, 0x28, 0xc42011e270, 0x28, 0x0, 0x0)
            /home/ubuntu/gopath/src/github.com/pilosa/pilosa/cmd/pilosa/main_test.go:332 +0xc9
    github.com/pilosa/pilosa/cmd/pilosa_test.TestMain_Set_Quick.func1(0xc420110000, 0x112, 0x112, 0x0)
            /home/ubuntu/gopath/src/github.com/pilosa/pilosa/cmd/pilosa/main_test.go:36 +0x222
    reflect.Value.call(0x87ffc0, 0xc420011fa0, 0x13, 0x91d319, 0x4, 0xc4200d70a0, 0x1, 0x1, 0xc4200a1500, 0x7f685d452000, ...)
            /home/ubuntu/go/src/reflect/value.go:434 +0x91f
    reflect.Value.Call(0x87ffc0, 0xc420011fa0, 0x13, 0xc4200d70a0, 0x1, 0x1, 0xc420017fb0, 0x0, 0x0)
            /home/ubuntu/go/src/reflect/value.go:302 +0xa4
    testing/quick.Check(0x87ffc0, 0xc420011fa0, 0xc420028f78, 0x0, 0xc42001a0d8)
            /home/ubuntu/go/src/testing/quick/quick.go:285 +0x28d
    github.com/pilosa/pilosa/cmd/pilosa_test.TestMain_Set_Quick(0xc42005b1e0)
            /home/ubuntu/gopath/src/github.com/pilosa/pilosa/cmd/pilosa/main_test.go:88 +0xcb
    testing.tRunner(0xc42005b1e0, 0x93bc88)
            /home/ubuntu/go/src/testing/testing.go:657 +0x96
    created by testing.(*T).Run
            /home/ubuntu/go/src/testing/testing.go:697 +0x2ca
    FAIL	github.com/pilosa/pilosa/cmd/pilosa	0.015s

### Information about your environment (OS/architecture, CPU, RAM, cluster/solo, configuration, etc.)

Ubuntu 16.04.1 LTS on amd64 in EC2

--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ glide.lock: glide glide.yaml
 vendor-update: glide.lock
 
 test: vendor
-	go test $(shell cd $(GOPATH)/src/$(CLONE_URL); go list ./... | grep -v vendor)
+	go test $(shell cd $(GOPATH)/src/$(CLONE_URL); go list ./... | grep -v vendor) $(TESTFLAGS)
 
 pilosa: vendor
 	go build -ldflags $(LDFLAGS) $(FLAGS) $(CLONE_URL)/cmd/pilosa

--- a/cmd/server_test.go
+++ b/cmd/server_test.go
@@ -23,6 +23,7 @@ import (
 
 	"github.com/pilosa/pilosa"
 	"github.com/pilosa/pilosa/cmd"
+	_ "github.com/pilosa/pilosa/test"
 )
 
 func TestServerHelp(t *testing.T) {

--- a/pilosa_test.go
+++ b/pilosa_test.go
@@ -4,6 +4,7 @@ import (
 	"testing"
 
 	"github.com/pilosa/pilosa"
+	_ "github.com/pilosa/pilosa/test"
 )
 
 func TestValidateName(t *testing.T) {

--- a/pql/parser_test.go
+++ b/pql/parser_test.go
@@ -19,6 +19,7 @@ import (
 	"testing"
 
 	"github.com/pilosa/pilosa/pql"
+	_ "github.com/pilosa/pilosa/test"
 )
 
 // Ensure the parser can parse PQL.

--- a/roaring/roaring_test.go
+++ b/roaring/roaring_test.go
@@ -25,6 +25,7 @@ import (
 	"testing/quick"
 
 	"github.com/pilosa/pilosa/roaring"
+	_ "github.com/pilosa/pilosa/test"
 )
 
 // Ensure an empty bitmap returns false if checking for existence.

--- a/server.go
+++ b/server.go
@@ -56,6 +56,7 @@ type Server struct {
 
 	// Cluster configuration.
 	// Host is replaced with actual host after opening if port is ":0".
+	Network string
 	Host    string
 	Cluster *Cluster
 
@@ -80,6 +81,8 @@ func NewServer() *Server {
 		Broadcaster:       NopBroadcaster,
 		BroadcastReceiver: NopBroadcastReceiver,
 
+		Network: "tcp",
+
 		AntiEntropyInterval: DefaultAntiEntropyInterval,
 		PollingInterval:     DefaultPollingInterval,
 		MetricInterval:      0,
@@ -103,7 +106,7 @@ func (s *Server) Open() error {
 	}
 
 	// Open HTTP listener to determine port (if specified as :0).
-	ln, err := net.Listen("tcp", ":"+port)
+	ln, err := net.Listen(s.Network, ":"+port)
 	if err != nil {
 		return fmt.Errorf("net.Listen: %v", err)
 	}

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -36,6 +36,7 @@ import (
 	"github.com/pilosa/pilosa"
 	"github.com/pilosa/pilosa/httpbroadcast"
 	"github.com/pilosa/pilosa/server"
+	"github.com/pilosa/pilosa/test"
 )
 
 // Ensure program can process queries and maintain consistency.
@@ -542,6 +543,7 @@ func NewMain() *Main {
 	}
 
 	m := &Main{Command: server.NewCommand(os.Stdin, os.Stdout, os.Stderr)}
+	m.Server.Network = *test.Network
 	m.Config.DataDir = path
 	m.Config.Host = "localhost:0"
 	m.Command.Stdin = &m.Stdin
@@ -580,7 +582,10 @@ func (m *Main) Reopen() error {
 	// Create new main with the same config.
 	config := m.Config
 	m.Command = server.NewCommand(os.Stdin, os.Stdout, os.Stderr)
+	m.Server.Network = *test.Network
 	m.Config = config
+
+	println("dbg/network", *test.Network)
 
 	// Run new program.
 	if err := m.Run(); err != nil {

--- a/statsd/statsd_test.go
+++ b/statsd/statsd_test.go
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"github.com/pilosa/pilosa/statsd"
+	_ "github.com/pilosa/pilosa/test"
 )
 
 func TestStatsClient_WithTags(t *testing.T) {

--- a/test/test.go
+++ b/test/test.go
@@ -1,0 +1,8 @@
+package test
+
+import "flag"
+
+// Test flags.
+var (
+	Network = flag.String("network", "tcp", "network name")
+)

